source/feature-documentation/core/nodes.rst

```diff
@@ -0,0 +1,277 @@
+Nodes
+=====
+
+Meshroom uses a nodes to control the reconstruction.
+A node represents a task that can be executed.
+
+Attributes
+++++++++++
+
+This is how we control what goes in the node and what comes out.
+There are many types of attributes that are used.
+
+Edges
++++++
+
+File attributes can be connected.
+This represents a dependency for that node.
+If the output of node A and node B is connected to the input of node C,
+C will not be executed until A and B are successful.
+This makes it easy to set up complicated workflows.
+
+Hashes
+++++++
+
+Every node has a hash based on its attributes.
+If an attribute that changes the output of the node is changed,
+that node will now have a different hash and so any previous computation done by this node will be invalid and so the progress is now gone.
+Any nodes that depend on this node will also change their hash because their file input attribute changed the directory which is based on the hash of the first node.
+And then that happens to any nodes that depend on those nodes and so on.
+Since all of the data is still stored in the cache folder under the previous hash,
+no time is lost if the attribute is changed back to the first value because the new hash will match the first hash.
+
+Files
++++++
+
+Every node has associated log(s), status(es) and statistics files.
+These allow the progress and performance of the process to be monitored.
+
+API
```

Comment:
That´s great. We should provide some examples somewhere similar to what I attempted here https://github.com/natowi/meshroom_external_plugins/wiki/2-DevDoc